Assets/_BForBoss/_Weapons/Scripts/Weapons/MeleeWeaponBehaviour.cs

```diff
@@ -0,0 +1,101 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Perigon.Weapons
+{
+    
+    public interface IMeleeWeapon
+    {
+        float CurrentCooldown { get; }
+        float MaxCooldown { get; }
+        bool CanMelee { get; }
+    }
+    
+    public class MeleeWeaponBehaviour : MonoBehaviour, IMeleeWeapon
+    {
+        [SerializeField] private MeleeScriptableObject _meleeScriptable;
+        [SerializeField] private Transform _playerTransform;
+
+        [SerializeField] private bool _canAttackMany = true;
+
+        private MeleeWeapon _weapon;
+        private InputAction _meleeActionInputAction;
+
+        public float CurrentCooldown => _weapon?.CurrentCooldown ?? 0f;
+        public float MaxCooldown => _meleeScriptable != null ? _meleeScriptable.AttackCoolDown : 1f;
+        public bool CanMelee => _weapon?.CanMelee ?? false;
+
+        public void Initialize(InputAction meleeAttackAction, IMeleeProperties properties = null)
+        {
+            _meleeActionInputAction = meleeAttackAction;
+            _weapon = new MeleeWeapon(properties ?? _meleeScriptable);
+            BindActions();
+        }
+
+        private void OnMeleeInputAction(InputAction.CallbackContext context)
+        {
+            if (context.performed)
+            {
+                var t = _playerTransform ? _playerTransform : transform;
+                if(_canAttackMany)
+                    _weapon.AttackManyIfPossible(t.position, t.forward);
+                else
+                    _weapon.AttackOneIfPossible(t.position, t.forward);
+            }
+        }
+
+        private void BindActions()
+        {
+            _meleeActionInputAction.performed += OnMeleeInputAction;
+            _meleeActionInputAction.canceled += OnMeleeInputAction;
+        }
+
+        private void Awake()
+        {
+            if (_playerTransform == null)
+            {
+                Debug.LogWarning("Player transform is not set on melee weapon, melee attacks may not work as expected");
+            }
+        }
+
+        private void Update()
+        {
+            _weapon.DecrementCooldown(Time.deltaTime);
+        }
+
+        private void OnEnable()
+        {
+            if (_meleeActionInputAction != null)
+            {
+                BindActions();
+            }
+        }
+
+        private void OnDisable()
+        {
+            if (_meleeActionInputAction != null)
+            {
+                _meleeActionInputAction.performed -= OnMeleeInputAction;
+                _meleeActionInputAction.canceled -= OnMeleeInputAction;
+            }
+        }
+
+        private void OnDrawGizmosSelected()
+        {
+            if (_meleeScriptable != null)
+            {
+                var t = _playerTransform ? _playerTransform : transform;
```

Comment:
Can you post a scren shot of the Gizmo in action?

Reply:
Sure, I'll post that tonight

Reply:
Added to the description!